### PR TITLE
Expect some progress in centaur tests

### PR DESCRIPTION
**Pull request by cjllanwarne**
Adds an expectation before all `pollUntilStatus` tests that we see at least *some* progress within a minute

### CODE REVIEW

File: src/ci/resources/centaur_application.conf [hunk 1]
```diff
@@ -11,7 +11,7 @@ centaur {
   standardTestCasePath: ${CROMWELL_BUILD_CENTAUR_TEST_DIRECTORY}
 
   error-reporter {
-    retry-attempts: 2
+    retry-attempts: 0
```
Comment (salonishah11) [inline src/ci/resources/centaur_application.conf]: Will this increase the occurrences of flaky tests in Travis?

File: centaur/src/main/scala/centaur/test/Test.scala [hunk 2]
```diff
@@ -171,10 +171,50 @@ object Operations {
   }
 
   /**
-    * Polls until a specific status is reached. If a terminal status which wasn't expected is returned, the polling
-    * stops with a failure.
+    * Polls until a valid status is reached.
+    * If an unexpected status returned, the polling stops with a failure.
```
Comment (mcovarr) [inline centaur/src/main/scala/centaur/test/Test.scala]: doesn't quite sentence

File: src/ci/resources/centaur_application.conf (hunk 1 above)
Comment (mcovarr) [reply src/ci/resources/centaur_application.conf]: did you mean to commit this?

File: centaur/src/main/scala/centaur/test/Test.scala [hunk 3]
```diff
@@ -171,10 +171,50 @@ object Operations {
...
-  def pollUntilStatus(workflow: SubmittedWorkflow, testDefinition: Workflow, expectedStatus: WorkflowStatus): Test[SubmittedWorkflow] = {
+  def expectSomeProgress(workflow: SubmittedWorkflow,
+                         testDefinition: Workflow,
+                         expectedStatuses: Set[WorkflowStatus],
+                         timeout: FiniteDuration): Test[SubmittedWorkflow] = {
+    new Test[SubmittedWorkflow] {
+      def status(remainingTimeout: FiniteDuration): IO[SubmittedWorkflow] = {
+        for {
+          workflowStatus <- CentaurCromwellClient.status(workflow)
+          mappedStatus <- workflowStatus match {
```
Comment (aednichols) [inline centaur/src/main/scala/centaur/test/Test.scala]: Any reason you don't do `mappedStatus = workflowStatus` and get rid of all the non-IO `IO`'s in the `match`?

Comment (aednichols) [reply centaur/src/main/scala/centaur/test/Test.scala]: Is it because the `IO.sleep` is captured in the IO and deferred?

**Pull request merged**